be/src/vec/exprs/vexpr.cpp

```diff
@@ -722,23 +735,41 @@ Status VExpr::_evaluate_inverted_index(VExprContext* context, const FunctionBase
         }
     }
 
+    auto result_bitmap = segment_v2::InvertedIndexResultBitmap();
+    auto set_result = [&]() {
+        if (!result_bitmap.is_empty()) {
+            index_context->set_inverted_index_result_for_expr(this, result_bitmap);
+            for (int column_id : column_ids) {
+                index_context->set_true_for_inverted_index_status(this, column_id);
+            }
+        }
+    };
+
+    if (!variant_column_exist) {
```

Comment:
why is variant special here, what if it is a normal column and not exist in segment

Reply:
At first, I didn't account for missing regular fields either. After testing, I found that there's still that extra 2-second delay when regular fields don't exist. Let me see if there's a more general fix.